src/arvgcregisternode.c

```diff
@@ -482,13 +485,16 @@ _read_from_port (ArvGcRegisterNode *gc_register_node, gint64 address, gint64 len
 		memcpy (cache, buffer, length);
 	}
 
-	arv_gc_port_read (ARV_GC_PORT (port), buffer, address, length, &local_error);
-	if (local_error == NULL)
-		cachable = _get_cachable (gc_register_node, &local_error);
-	if (local_error != NULL) {
-		g_propagate_error (error, local_error);
-		gc_register_node->cached = FALSE;
-		return;
+	if (!cached || cache_policy == ARV_REGISTER_CACHE_POLICY_DEBUG) {
+		arv_gc_port_read (ARV_GC_PORT (port), buffer, address, length, &local_error);
+		if (local_error != NULL) {
+			g_propagate_error(error, local_error);
+			gc_register_node->cached = FALSE;
+			if (cached && cache_policy == ARV_REGISTER_CACHE_POLICY_DEBUG) {
```

Comment:
It looks like this statement `cached && cache_policy == ARV_REGISTER_CACHE_POLICY_DEBUG` is always going to be evaluated to false since it is placed inside another if at the line 488 
`if (!cached || cache_policy == ARV_REGISTER_CACHE_POLICY_DEBUG)`

Reply:
| cached | policy==DEBUG | outer | inner |
|--------|---------------|-------|-------|
|  false |     false     | true  | false |
|  true  |     false     | false | false |
|  false |     true      | true | false |
|  true  |     true      | true  | true  |